apps/calculation/calculation.cpp

```diff
@@ -60,19 +60,44 @@ KDCoordinate Calculation::height(Context * context) {
   if (m_height < 0) {
     Layout inputLayout = createInputLayout();
     KDCoordinate inputHeight = inputLayout.layoutSize().height();
+    KDCoordinate inputWidth = inputLayout.layoutSize().width();
     Layout approximateLayout = createApproximateOutputLayout(context);
     Layout exactLayout = createExactOutputLayout();
+    float smallMargin = 2 * Metric::CommonSmallMargin;
+    float lessSmallMargin = 4 * Metric::CommonSmallMargin;
+    int maxWidth = 290;
+    bool singleLine = false;
     if (shouldOnlyDisplayExactOutput()) {
       KDCoordinate exactOutputHeight = exactLayout.layoutSize().height();
-      m_height = inputHeight+exactOutputHeight;
+      KDCoordinate exactOutputWidth = exactLayout.layoutSize().width();
+      singleLine = exactOutputWidth + inputWidth < maxWidth - 40;
+      if (singleLine) {
+        m_height = (inputHeight >= exactOutputHeight) ? inputHeight + smallMargin : exactOutputHeight + smallMargin;
+      } else {
+        m_height = inputHeight + exactOutputHeight + lessSmallMargin;
+      }
     } else if (shouldOnlyDisplayApproximateOutput(context)) {
       KDCoordinate approximateOutputHeight = approximateLayout.layoutSize().height();
-      m_height = inputHeight+approximateOutputHeight;
+      KDCoordinate approximateOutputWidth = approximateLayout.layoutSize().width();
+      singleLine = approximateOutputWidth + inputWidth < maxWidth - 40;
+      if (singleLine) {
+        m_height = (inputHeight >= approximateOutputHeight) ? inputHeight + smallMargin : approximateOutputHeight + smallMargin;
+      } else {
+        m_height = inputHeight + approximateOutputHeight + lessSmallMargin;
+      }
     } else {
       KDCoordinate approximateOutputHeight = approximateLayout.layoutSize().height();
       KDCoordinate exactOutputHeight = exactLayout.layoutSize().height();
       KDCoordinate outputHeight = max(exactLayout.baseline(), approximateLayout.baseline()) + max(exactOutputHeight-exactLayout.baseline(), approximateOutputHeight-approximateLayout.baseline());
-      m_height = inputHeight + outputHeight;
+      KDCoordinate exactOutputWidth = exactLayout.layoutSize().width();
+      KDCoordinate approximateOutputWidth = approximateLayout.layoutSize().width();
+      KDCoordinate outputWidth = exactOutputWidth + approximateOutputWidth;
+      singleLine = outputWidth + inputWidth < maxWidth - 70;
```

Comment:
This should fix the bug with 11^9
```suggestion
      singleLine = outputWidth + inputWidth < maxWidth - 40;
```